heron/api/src/java/com/twitter/heron/streamlet/Config.java

```diff
@@ -109,19 +148,24 @@ public Builder setUserConfig(String key, Object value) {
     }
 
     /**
-     * Sets the topology to use the Kryo serializer for serializing
+     * Sets the topology to use the specified serializer for serializing
      * streamlet elements
+     * @param topologySerializer The serializer to be used in this topology
      */
-    public Builder useKryoSerializer() {
-      try {
-        config.setSerializationClassName(new KryoSerializer().getClass().getName());
-      } catch (NoClassDefFoundError e) {
-        throw new RuntimeException("Linking with kryo is needed because useKryoSerializer is used");
+    public Builder setSerializer(Serializer topologySerializer) {
+      serializer = topologySerializer;
+      if (serializer == Serializer.KRYO) {
+        try {
+          config.setSerializationClassName(new KryoSerializer().getClass().getName());
+        } catch (NoClassDefFoundError e) {
+          throw new RuntimeException("Linking with kryo is needed because setSerializer is used. " +
+              "You may be attempting to set the serializer more than once.");
```

Comment:
Not sure if this error message is accurate or not.

Reply:
You’re right. I was misidentifying the problem. Using KryoSerializer.class instead of instantiating a KryoSerializer seems to have fixed the issue. You can still set the serializer more than once but again, I believe this is an issue in all builder pattern classes.